libsql/src/database.rs

```diff
@@ -358,11 +378,31 @@ impl Database {
             }
 
             #[cfg(feature = "replication")]
-            DbType::Sync { db } => {
+            DbType::Sync { db, encryption_key } => {
                 use crate::local::impls::LibsqlConnection;
 
                 let conn = db.connect()?;
 
+                if !cfg!(feature = "encryption-at-rest") {
+                    let _ = encryption_key;
+                    tracing::warn!("Encryption at rest is not enabled, ignoring encryption_key");
+                }
+                #[cfg(feature = "encryption-at-rest")]
+                if let Some(encryption_key) = encryption_key {
+                    let rc = unsafe {
```

Comment:
We should put all the unsafe code into the sys crate or with in the local module.